olp-cpp-sdk-core/src/network2/winhttp/NetworkProtocolWinHttp.cpp

```diff
@@ -0,0 +1,1063 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "NetworkProtocolWinHttp.h"
+
+#include <cassert>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "olp/core/logging/Log.h"
+#include "olp/core/network2/Network.h"
+#include "olp/core/network2/NetworkRequest.h"
+#include "olp/core/network2/NetworkResponse.h"
+#include "olp/core/network2/NetworkTypes.h"
+#include "olp/core/porting/make_unique.h"
+
+namespace {
+
+constexpr int kNetworkUncompressionChunkSize = 1024 * 16;
+
+LPCSTR
+errorToString(DWORD err) {
+  switch (err) {
+    case ERROR_NOT_ENOUGH_MEMORY:
+      return "Out of memory";
+    case ERROR_WINHTTP_CANNOT_CONNECT:
+      return "Cannot connect";
+    case ERROR_WINHTTP_CHUNKED_ENCODING_HEADER_SIZE_OVERFLOW:
+      return "Parsing overflow";
+    case ERROR_WINHTTP_CLIENT_AUTH_CERT_NEEDED:
+      return "Authentication required";
+    case ERROR_WINHTTP_CONNECTION_ERROR:
+      return "Connection error";
+    case ERROR_WINHTTP_HEADER_COUNT_EXCEEDED:
+      return "Header count exceeded";
+    case ERROR_WINHTTP_HEADER_SIZE_OVERFLOW:
+      return "Header size overflow";
+    case ERROR_WINHTTP_INCORRECT_HANDLE_STATE:
+      return "Invalid handle state";
+    case ERROR_WINHTTP_INCORRECT_HANDLE_TYPE:
+      return "Invalid handle type";
+    case ERROR_WINHTTP_INTERNAL_ERROR:
+      return "Internal error";
+    case ERROR_WINHTTP_INVALID_SERVER_RESPONSE:
+      return "Invalid server response";
+    case ERROR_WINHTTP_INVALID_URL:
+      return "Invalid URL";
+    case ERROR_WINHTTP_LOGIN_FAILURE:
+      return "Login failed";
+    case ERROR_WINHTTP_NAME_NOT_RESOLVED:
+      return "Name not resolved";
+    case ERROR_WINHTTP_OPERATION_CANCELLED:
+      return "Cancelled";
+    case ERROR_WINHTTP_REDIRECT_FAILED:
+      return "Redirect failed";
+    case ERROR_WINHTTP_RESEND_REQUEST:
+      return "Resend request";
+    case ERROR_WINHTTP_RESPONSE_DRAIN_OVERFLOW:
+      return "Response overflow";
+    case ERROR_WINHTTP_SECURE_FAILURE:
+      return "Security error";
+    case ERROR_WINHTTP_TIMEOUT:
+      return "Timed out";
+    case ERROR_WINHTTP_UNRECOGNIZED_SCHEME:
+      return "Invalid scheme";
+  }
+  return "Unknown error";
+}
+
+olp::network2::ErrorCode errorToCode(DWORD err) {
```

Comment:
```suggestion
olp::network2::ErrorCode ErrorToCode(DWORD err) {
```
Or even better to emphasize which Error and which Code:
```suggestion
olp::network2::ErrorCode WinErrorToNetwork2ErrorCode(DWORD err) {
```